operatorcourier/flatten.py

```diff
@@ -61,34 +63,56 @@ def get_flattened_files_info(source_dir: str) -> [(str, str)]:
     return file_paths_to_copy
 
 
-def parse_version_folder(base_dir: str, version_folder_name: str,
+def get_folder_semver(base_dir: str, version_folder_name: str):
+    version_folder_path = os.path.join(base_dir, version_folder_name)
+
+    for item in os.listdir(version_folder_path):
+        item_path = os.path.join(version_folder_path, item)
+        if not os.path.isfile(item_path) or not is_yaml_file(item_path):
+            continue
+
+        with open(item_path, 'r') as f:
+            file_content = f.read()
+
+        if identify.get_operator_artifact_type(file_content) == 'ClusterServiceVersion':
+            try:
+                csv_version = safe_load(file_content)['spec']['version']
+            except MarkedYAMLError:
+                msg = f'{item} is not a valid YAML file.'
+                logger.error(msg)
+                raise errors.OpCourierBadYaml(msg)
+            except KeyError:
+                msg = f'{item} is not a valid CSV file as "spec.version" ' \
+                      f'field is required'
+                logger.error(msg)
+                raise errors.OpCourierBadBundle(msg, {})
+            return csv_version
+
+    msg = 'This version directory does not contain any valid CSV file.'
+    logger.error(msg)
+    raise errors.OpCourierBadBundle(msg, {})
```

Comment:
If we're already going through this trouble, can't we just return nil here? Why do we need to throw an error if we have some extra folders we don't care about?